components/sidebar/sidebar_channel/index.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+
+import {GlobalState} from 'mattermost-redux/types/store';
+import {GenericAction} from 'mattermost-redux/types/actions';
+import {makeGetChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getUser} from 'mattermost-redux/selectors/entities/users';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+
+import {Constants} from 'utils/constants';
+
+import SidebarChannel from './sidebar_channel';
+
+type OwnProps = {
+    channelId: string;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    const getChannel = makeGetChannel();
+    const channel = getChannel(state, {id: ownProps.channelId});
+    const currentTeam = getCurrentTeam(state);
+    let teammate;
+
+    if (channel.type === Constants.DM_CHANNEL && channel.teammate_id) {
+        teammate = getUser(state, channel.teammate_id);
```

Comment:
Nothing to change here right now, but I'm thinking it might be a decent idea to break up this component further so that we might have a `SidebarChannel` component, but that renders a `SidebarDirectChannel` or `SidebarGroupChannel` which will contain the special handling for those channels. That way, we can have the `SidebarChannel` get the team and channel while the group and DM channel components can pull whatever extra data they need from the store on their own.

Reply:
I can break up the components now, I'll make that change.